mlflow/pyfunc/__init__.py

```diff
@@ -716,6 +717,24 @@ def load_model(
     return PyFuncModel(model_meta=model_meta, model_impl=model_impl)
 
 
+def print_model_dependencies(model_uri):
+    """
+    Given a model URL, print model dependencies provided by model’s requirements.txt file.
+    And for databricks runtime, also print an instruction to update python environment with
+    the model dependencies.
+    """
+    local_path = _download_artifact_from_uri(artifact_uri=model_uri, output_path=None)
+    req_file_path = os.path.join(local_path, _REQUIREMENTS_FILE_NAME)
+    print(f"model {model_uri} dependencies:")
+    with open(req_file_path, "r") as f:
+        print(f.read())
+    print("\n")
+    if is_in_databricks_runtime():
+        print(f"On databricks notebook, you can run command '%pip install -r {req_file_path}' "
+              "to update current python environment with model’s requirements.txt, and rerun "
+              "inference commands.")
```

Comment:
Instead of printing out dependencies, can we return them as a list of strings?

## Example use cases

```python
deps = list_model_dependencies(mdoel_uri)

# Check dependencies
print(deps)

# Construct pip install command
print("pip install", " ".join(deps))

# Copy dependencies
mlflow.*.log_model(..., pip_requirements=deps)
```

Reply:
we can.
and according to @mengxr 's suggestion, we should also print an instruction to tell user how to update python environment with these dependencies.